### PR TITLE
feat: Add memory session manager

### PR DESCRIPTION
**Pull request by DanielRivers**
# Explain your changes

This adds an implementation of a session manager with configuration.  

Currently there is a single manager of `memoryStorage`, more will be added including cookie and session.

The memory manager keys have a default set of keys of `accessToken`, `idToken` and `refreshToken`, this can be added to by passing in an enum as a generic when setting the storage type

# Checklist

- [x] I have read the [“Pull requests” section](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md#pull-requests) in the [contributing guidelines](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I agree to the terms within the [code of conduct](https://github.com/kinde-oss/.github/blob/main/.github/CODE_OF_CONDUCT.md).

🛟 _If you need help, consider asking for advice over in the [Kinde community](https://thekindecommunity.slack.com)._

### CODE REVIEW

File: lib/sessionManager/types.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+type Awaitable<T> = Promise<T>;
+
+export enum StorageKeys {
+  accessToken = "accessToken",
+  idToken = "idToken",
+  refreshToken = "refreshToken",
+}
+
+export type StorageSettingsType = {
```
Comment (peterphanouvong) [inline lib/sessionManager/types.ts:14]: What is the plan for this type 😃

File: package.json [hunk 2]
```diff
@@ -22,20 +22,17 @@
...
-    "prettier": "^3.2.5",
-    "typescript": "^5.4.5",
-    "typescript-eslint": "^7.9.0",
-    "vite": "^5.2.10",
-    "vite-plugin-dts": "^3.9.0",
-    "vitest": "^1.5.2"
-  },
-  "dependencies": {
-    "jose": "^5.2.4"
+    "@eslint/js": "^9.9.1",
```
Comment (peterphanouvong) [inline package.json:25]: Just for my own understanding - why have these been moved to dependencies?

**Review by peterphanouvong — APPROVED** [review]

File: lib/sessionManager/types.ts (hunk 1 above)
Comment (DanielRivers) [reply lib/sessionManager/types.ts:14]: there was a mistake in the `sessionManager/index.ts` it should have been using this type 👍

File: package.json (hunk 2 above)
Comment (DanielRivers) [reply package.json:25]: This change indicator is misleading, some dev dependancies were updated and I removed jose as it was not required.

Nothing else changed here.

**Pull request merged**